src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -36,22 +44,33 @@ def __init__(self, spiffe_socket: str = None) -> None:
                 'SPIFFE socket argument or environment variable invalid in DefaultWorkloadApiClient.'
             )
 
-    def get_spiffe_endpoint_socket(self) -> str:
-        """Returns the spiffe endpoint socket config for this WorkloadApiClient.
-
-        Returns:
-            str: spiffe endpoint socket configuration value.
-        """
-
-        return self._config.spiffe_endpoint_socket
+        self._spiffe_workload_api_stub = workload_pb2_grpc.SpiffeWorkloadAPIStub(
+            self._get_spiffe_grpc_channel()
+        )
 
     def fetch_x509_svid(self) -> X509Svid:
         """Fetches a SPIFFE X.509-SVID.
 
         Returns:
             X509Svid: Instance of X509Svid object.
         """
+        response = self._spiffe_workload_api_stub.FetchX509SVID(
+            workload_pb2.X509SVIDRequest()
+        )
+
+        item = next(response)
+        svid = item.svids[0]
```

Comment:
Do we need to validate `svids` is not None?

Reply:
yes, I left out some error handling as to keep the PR simple, but now it's probably worth adding it and demonstrate how to test it as well using the the FakeWorkloadApi. I'll include it in this PR.

Reply:
Fair enough, sorry for the comments on stuff you already planned on doing. :-)